src/actions/socialLogin/FacebookActions.js

```diff
@@ -0,0 +1,72 @@
+import axios from 'axios';
+
+import {
+  FACEBOOK_AUTHENTICATION,
+  FACEBOOK_AUTH_SUCCESS,
+  FACEBOOK_AUTH_FAIL,
+} from '../socialLoginTypes';
+
+/**
+ * This is a function which returns the facebookSuccess action
+ * @param {object} - A object parameter type*
+ * @return {none}
+ * @example
+ * foo = () => {}
+ */
+const facebookSuccess = resp => ({
+  type: FACEBOOK_AUTH_SUCCESS,
+  payload: resp.data.user,
+});
+
+/**
+ * This is a function which returns the facebookFail action
+ * @param {object} - A object parameter type*
+ * @return {none}
+ * @example
+ * foo = () => {}
+ */
+const facebookFail = error => ({
+  type: FACEBOOK_AUTH_FAIL,
+  payload: error.response.data.errors,
+});
+
+/**
+ * This is a function which sends facebook registration token to the api
+ * @param {string} - A string param
+ * @return {none}
+ * @example
+ * foo = (string) => {}
+ */
+const FacebookLogin = token => (dispatch) => {
+  dispatch({
+    type: FACEBOOK_AUTHENTICATION,
+  });
+  // @type {object}
+  const config = {
+    headers: { 'Content-Type': 'application/json' },
+    responseType: 'application/json',
+  };
+  // @type {object}
+  const userData = {
+    access_token: token,
+  };
+  return axios
+    .post(
+      'https://ah-backend-kronos-staging.herokuapp.com/api/social-auth/facebook/',
+      userData,
+      config,
+    )
+    .then((resp) => {
+      sessionStorage.setItem('token', resp.data.user.token);
+      sessionStorage.setItem('user_email', resp.data.user.email);
```

Comment:
What is the use of the email in the session storage?

Reply:
@Yiga-fred the `email` is what I use to display on the success page to re-affirm that the email matches the one used with respective social media account.